# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/webidl/ReadableStream.webidl
# Commit: 19547ae8dc19
# Full Hash: 19547ae8dc19492947d8b2100fc1df793f2fce6e
# Author: Tom Schuster <evilpies@gmail.com>
# Date: 2022-12-21 21:21:23
# Regressor Bug: 1734244
# File Overlap Count: 1
# Description:
#   Bug 1734244 - Implement async iteration of ReadableStream. r=mgaudet,peterv
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D153312
# ==============================================================================

diff -r a186000b3a89 -r 19547ae8dc19 dom/webidl/ReadableStream.webidl
--- a/dom/webidl/ReadableStream.webidl	Tue Dec 20 16:11:53 2022 +0000
+++ b/dom/webidl/ReadableStream.webidl	Tue Dec 20 16:26:25 2022 +0000
@@ -1,3 +1,12 @@
+/* -*- Mode: IDL; tab-width: 2; indent-tabs-mode: nil; c-basic-offset: 2 -*- */
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/.
+ *
+ * The origin of this IDL file is
+ * https://streams.spec.whatwg.org/#rs-class-definition
+ */
+
 [Exposed=*,
 //Transferable See Bug 1562065
 ]
@@ -22,8 +31,8 @@
   [Throws]
   sequence<ReadableStream> tee();
 
-  // Bug 1734244
-  // async iterable<any>(optional ReadableStreamIteratorOptions options = {});
+  [GenerateReturnMethod]
+  async iterable<any>(optional ReadableStreamIteratorOptions options = {});
 };
 
 enum ReadableStreamReaderMode { "byob" };
@@ -32,6 +41,10 @@
   ReadableStreamReaderMode mode;
 };
 
+dictionary ReadableStreamIteratorOptions {
+  boolean preventCancel = false;
+};
+
 dictionary ReadableWritablePair {
   required ReadableStream readable;
   required WritableStream writable;